lib/api/core/validation/index.js

```diff
@@ -1,690 +1,682 @@
+            if (!filters.length > 0 || filters[0] !== collectionSpec.validators) {
+              isValid = false;
+              manageErrorMessage('document', errorMessages, 'The document does not match validation filters.', verbose);
+            }
+          }
         }
-      }
-    }
 
-    return {
-      errorMessages: errorMessages,
-      valid: isValid
-    };
-  })
-  .then(validationObject => {
-    if (!verbose) {
-      // We only modify the request if the validation succeeds
-      if (collectionSpec.fields && collectionSpec.fields.children) {
-        request.input.body = this.recurseApplyDefault(isUpdate, request.input.body, collectionSpec.fields.children);
-      }
+        return {
+          errorMessages,
+          valid: isValid
+        };
+      })
+      .then(validationObject => {
+        if (!verbose) {
+          // We only modify the request if the validation succeeds
+          if (collectionSpec.fields && collectionSpec.fields.children) {
+            request.input.body = this.recurseApplyDefault(isUpdate, request.input.body, collectionSpec.fields.children);
+          }
 
-      return request;
-    }
+          return request;
+        }
 
-    return validationObject;
-  });
-};
+        return validationObject;
+      });
+  }
 
-/**
- * @param {boolean} isUpdate
- * @param {*} documentSubset
- * @param {...StructuredFieldSpecification} collectionSpecSubset
- */
-Validation.prototype.recurseApplyDefault = function validationRecurseApplyDefault (isUpdate, documentSubset, collectionSpecSubset) {
-  Object.keys(collectionSpecSubset).forEach(fieldName => {
-    var i;
-
-    if (
-      documentSubset.hasOwnProperty(fieldName) &&
-      this.types[collectionSpecSubset[fieldName].type].allowChildren &&
-      collectionSpecSubset[fieldName].children
-    ) {
-      if (Array.isArray(documentSubset[fieldName])) {
-        for (i = 0; i < documentSubset[fieldName].length; i++) {
-          documentSubset[fieldName][i] = this.recurseApplyDefault(isUpdate, documentSubset[fieldName][i], collectionSpecSubset[fieldName].children);
+  /**
+   * @param {boolean} isUpdate
+   * @param {*} documentSubset
+   * @param {...StructuredFieldSpecification} collectionSpecSubset
+   */
+  recurseApplyDefault(isUpdate, documentSubset, collectionSpecSubset) {
+    Object.keys(collectionSpecSubset).forEach(fieldName => {
+      const
+        specSubset = collectionSpecSubset[fieldName],
+        field = documentSubset[fieldName];
+
+      if (documentSubset.hasOwnProperty(fieldName) && this.types[specSubset.type].allowChildren && specSubset.children) {
+        if (Array.isArray(field)) {
+          for (let i = 0; i < field.length; i++) {
+            field[i] = this.recurseApplyDefault(isUpdate, field[i], specSubset.children);
+          }
+        }
+        else {
+          documentSubset[fieldName] = this.recurseApplyDefault(isUpdate, field, specSubset.children);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 101 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)